### PR TITLE
cache me api response in redis

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

**Comment by jacksonh** [comment]
I think we also might need to invalidate the cache when the user features change

**Comment by sywhb** [comment]
> I think we also might need to invalidate the cache when the user features change

Yeah that is a good one

**Comment by sywhb** [comment]
> I think we also might need to invalidate the cache when the user features change

Actually features are not cached because they are not queried from the function_resolver

File: packages/api/src/resolvers/function_resolvers.ts [hunk 1]
```diff
@@ -377,7 +381,14 @@ export const functionResolvers = {
         return undefined
       }
 
-      return findUserFeatures(ctx.claims.uid)
+      let features = await getCachedFeatures(ctx.claims.uid)
+      if (!features) {
+        features = await findUserFeatures(ctx.claims.uid)
+
+        await cacheFeatures(ctx.claims.uid, features)
```
Comment (pradeepmurugesan) [inline packages/api/src/resolvers/function_resolvers.ts:388]: Why do we need to `await` for the `caching` to be completed? Can't this happen in the asynchronously ?

Comment (jacksonh) [reply packages/api/src/resolvers/function_resolvers.ts:388]: generally we await on all our backend functions, one reason is it can better handle errors.

**Comment by jacksonh** [comment]
I will close this one as I think it didn't quite handle some of the edge cases we came up with.

**Pull request closed without merging**